ansible/roles/test/files/ptftests/vxlan-decap.py

```diff
@@ -0,0 +1,293 @@
+                         ip_dst=dst_ip,
+                       )
+        exp_packet = simple_tcp_packet(
+                         eth_dst=self.random_mac,
+                         eth_src=dst_mac,
+                         ip_src=src_ip,
+                         ip_dst=dst_ip,
+                         ip_ttl = 63,
+                       )
+
+        exp_packet = Mask(exp_packet)
+        exp_packet.set_do_not_care(0, 6*8)
+
+        for i in xrange(self.nr):
+            testutils.send_packet(self, acc_port, packet)
+        nr_rcvd = testutils.count_matched_packets_all_ports(self, exp_packet, pc_ports, timeout=0.1)
+        rv = rv and (nr_rcvd == self.nr)
+        return rv
+
+
+    def checkRegularFromPC(self, acc_port, net_port, test):
+        rv = True
+        src_mac = self.random_mac
+        dst_mac = self.dut_mac
+        src_ip = test['src_ip']
+        dst_ip = test['vlan_ip_prefix'] % acc_port
+
+        packet = simple_tcp_packet(
+                         eth_dst=dst_mac,
+                         eth_src=src_mac,
+                         ip_src=src_ip,
+                         ip_dst=dst_ip,
+                       )
+
+        exp_packet = simple_tcp_packet(
+                         eth_dst=self.ptf_mac_addrs['eth%d' % acc_port],
+                         eth_src=dst_mac,
+                         ip_src=src_ip,
+                         ip_dst=dst_ip,
+                         ip_ttl = 63,
+                       )
+
+        for i in xrange(self.nr):
+            testutils.send_packet(self, net_port, packet)
+        nr_rcvd = testutils.count_matched_packets(self, exp_packet, acc_port, timeout=0.1)
+        rv = rv and (nr_rcvd == self.nr)
+        return rv
+
+    def checkVxlan(self, acc_port, net_port, test):
+        rv = True
+        inner_dst_mac = self.ptf_mac_addrs['eth%d' % acc_port]
+        inner_src_mac = self.dut_mac
+        inner_src_ip = test['vlan_gw']
+        inner_dst_ip = test['vlan_ip_prefix'] % acc_port
+        dst_mac = self.dut_mac
+        src_mac = self.random_mac
+        ip_dst = self.loopback_ip
+
+        inpacket = simple_arp_packet(
+                         eth_dst=inner_dst_mac,
+                         eth_src=inner_src_mac,
+                         arp_op=2,
+                         ip_snd=inner_src_ip,
+                         ip_tgt=inner_dst_ip,
+                         hw_snd=inner_src_mac,
+                         hw_tgt=inner_dst_mac
+                       )
+
+        packet = simple_vxlan_packet(
+                       eth_dst=dst_mac,
+                       eth_src=src_mac,
+                       ip_src=test['src_ip'],
+                       ip_dst=ip_dst,
+                       vxlan_vni=test['vni'],
+                       inner_frame=inpacket
+                 )
+        for i in xrange(self.nr):
+            testutils.send_packet(self, net_port, packet)
+        nr_rcvd = testutils.count_matched_packets(self, inpacket, acc_port, timeout=0.1)
```

Comment:
Just checking if `timeout `value of `0.1` sufficient for all cases?

Reply:
I don't think ASIC will hold a packet for more then 0.1 seconds inside.